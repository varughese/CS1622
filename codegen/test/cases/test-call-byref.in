int g;
int garr[4];

void bar(int a[], int b, int c, int d)
{
    output(a[1]);
    a[2] = input();
    b = input();
    c = input();
    d = input();
}

void foo(int a[], int b, int c)
{
    output(a[1]);
    a[0] = input();
    a[1] = input();
    bar(a, a[1], b, c);
    b = input();
    c = input();
}

void main(void)
{
    int x;
    int y[4];

    x = 101;
    y[0] = 102;
    y[1] = 103;
    y[2] = 104;
    y[3] = 105;

    foo(y, x, y[3]);

    output(y[0]);
    output(y[1]);
    output(y[2]);
    output(y[3]);
    output(x);

    g = 201;
    garr[0] = 202;
    garr[1] = 203;
    garr[2] = 204;
    garr[3] = 205;

    foo(garr, g, garr[3]);

    output(garr[0]);
    output(garr[1]);
    output(garr[2]);
    output(garr[3]);
    output(g);
}
