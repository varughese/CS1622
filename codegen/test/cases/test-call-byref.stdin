11080
12080
13080
14080
15080
16080
17080
18080
21080
22080
23080
24080
25080
26080
27080
28080
