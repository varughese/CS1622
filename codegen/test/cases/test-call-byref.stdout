103
12080
11080
12080
13080
105
101
203
22080
21080
22080
23080
205
201
